src/domains/migration/hooks/use-migration-transactions.ts

```diff
@@ -12,60 +12,80 @@ import {
 } from "@/utils/polygon-migration";
 import { Migration } from "@/domains/migration/migration.contracts";
 
+export const fetchMigrationTransactions = async ({ profile, page }: { page: number; profile: Contracts.IProfile }) => {
+	const wallet = await profile.walletFactory().fromAddress({
+		address: migrationWalletAddress(),
+		coin: "ARK",
+		network: migrationNetwork(),
+	});
+
+	const senderIds = profile
+		.wallets()
+		.values()
+		.filter((wallet) => wallet.networkId() === migrationNetwork())
+		.map((wallet) => wallet.address());
+
+	if (senderIds.length === 0) {
+		return {
+			cursor: page,
+			hasMore: false,
+			items: [],
+		};
+	}
+
+	const query: {
+		recipientId: string;
+		senderId: string;
+		amount: { from: string };
+		fee: { from: string };
+		timestamp?: { from?: number; to?: number };
+		page?: number;
+	} = {
+		amount: { from: BigNumber.make(migrationMinBalance()).times(1e8).toString() },
+		fee: { from: BigNumber.make(migrationTransactionFee()).times(1e8).toString() },
+		page,
+		recipientId: migrationWalletAddress(),
+		senderId: senderIds.join(","),
+	};
+
+	if (polygonMigrationStartTime() > 0) {
+		query.timestamp = { from: polygonMigrationStartTime() };
+	}
+
+	const transactions = await wallet.transactionIndex().received(query);
+
+	return {
+		cursor: Number(transactions.currentPage()),
+		hasMore: transactions.items().length > 0,
+		items: transactions.items(),
+	};
+};
+
 export const useMigrationTransactions = ({ profile }: { profile: Contracts.IProfile }) => {
 	const { profileIsRestoring } = useConfiguration();
 	const { migrations, storeTransactions } = useMigrations();
 	const [latestTransactions, setLatestTransactions] = useState<DTO.ExtendedConfirmedTransactionData[]>([]);
 	const [isLoadingTransactions, setIsLoadingTransactions] = useState(true);
+	const [hasMore, setHasMore] = useState(false);
+	const [page, setPage] = useState(0);
+	const limit = 10;
```

Comment:
this limit needs to be applied to the API call, now you slice an array with 100 values and leave 90 out from the UI and when paginating you request another 100, etc